[BUG] datasourcev2_read_test failure on Databricks 14.3

On https://github.com/razajafri/spark-rapids/tree/SP-10661-db-14.3-deletion-vectors branch which is pointing to [commit](45c83d3852238f9343c7f36af6fa89286cceb508) at the time of writing this fails to execute datasourcev2_read_test when you run it by executing the following lines after building spark-rapids on Databricks 14.3 

`WITH_DEFAULT_UPSTREAM_SHIM=0 SPARK_SUBMIT_FLAGS="--conf spark.rapids.sql.detectDeltaLogQueries=false --conf spark.rapids.sql.format.parquet.reader.type=PERFILE" TEST_PARALLEL=0 TESTS=datasourcev2_read_test.py ./jenkins/databricks/test.sh` 

The following exception is thrown 

```
E                   py4j.protocol.Py4JJavaError: An error occurred while calling o195.load.
E                   : org.apache.spark.SparkClassNotFoundException: [DATA_SOURCE_NOT_FOUND] Failed to find data source: com.nvidia.spark.rapids.tests.datasourcev2.parquet.ArrowColumnarDataSourceV2. Please find packages at `https://spark.apache.org/third-party-projects.html`. SQLSTATE: 42K02
E                       at org.apache.spark.sql.errors.QueryExecutionErrors$.dataSourceNotFoundError(QueryExecutionErrors.scala:913)
E                       at org.apache.spark.sql.execution.datasources.DataSource$.lookupDataSource(DataSource.scala:789)
E                       at org.apache.spark.sql.execution.datasources.DataSource$.lookupDataSourceV2(DataSource.scala:857)
E                       at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:342)
E                       at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:231)
E                       at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
E                       at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
E                       at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
E                       at java.lang.reflect.Method.invoke(Method.java:498)
E                       at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
E                       at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:397)
E                       at py4j.Gateway.invoke(Gateway.java:306)
E                       at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
E                       at py4j.commands.CallCommand.execute(CallCommand.java:79)
E                       at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:199)
E                       at py4j.ClientServerConnection.run(ClientServerConnection.java:119)
E                       at java.lang.Thread.run(Thread.java:750)
E                   Caused by: java.lang.ClassNotFoundException: com.nvidia.spark.rapids.tests.datasourcev2.parquet.ArrowColumnarDataSourceV2.DefaultSource
E                       at java.net.URLClassLoader.findClass(URLClassLoader.java:387)
E                       at java.lang.ClassLoader.loadClass(ClassLoader.java:419)
E                       at java.lang.ClassLoader.loadClass(ClassLoader.java:352)
E                       at org.apache.spark.sql.execution.datasources.DataSource$.$anonfun$lookupDataSource$6(DataSource.scala:773)
E                       at scala.util.Try$.apply(Try.scala:213)
E                       at org.apache.spark.sql.execution.datasources.DataSource$.$anonfun$lookupDataSource$5(DataSource.scala:773)
E                       at scala.util.Failure.orElse(Try.scala:224)
E                       at org.apache.spark.sql.execution.datasources.DataSource$.lookupDataSource(DataSource.scala:773)
E                       ... 15 more
```

--- a/integration_tests/run_pyspark_from_build.sh
+++ b/integration_tests/run_pyspark_from_build.sh
@@ -287,41 +287,6 @@ else
 
     export PYSP_TEST_spark_hadoop_hive_exec_scratchdir="$RUN_DIR/hive"
 
-    # Extract Databricks version from deployed configs.
-    # spark.databricks.clusterUsageTags.sparkVersion is set automatically on Databricks
-    # notebooks but not when running Spark manually.
-    #
-    # At the OS level the DBR version can be obtailed via
-    # 1. DATABRICKS_RUNTIME_VERSION environment set by Databricks, e.g., 11.3
-    # 2. File at /databricks/DBR_VERSION created by Databricks, e.g., 11.3
-    # 3. The value for Spark conf in file /databricks/common/conf/deploy.conf created by Databricks,
-    #    e.g. 11.3.x-gpu-ml-scala2.12
-    #
-    # For cases 1 and 2 append '.' for version matching in 3XYdb SparkShimServiceProvider
-    #
-    DBR_VERSION=/databricks/DBR_VERSION
-    DB_DEPLOY_CONF=/databricks/common/conf/deploy.conf
-    if [[ -n "${DATABRICKS_RUNTIME_VERSION}" ]]; then
-      export PYSP_TEST_spark_databricks_clusterUsageTags_sparkVersion="${DATABRICKS_RUNTIME_VERSION}."
-    elif [[ -f $DBR_VERSION || -f $DB_DEPLOY_CONF ]]; then
-      DB_VER="$(< ${DBR_VERSION})." || \
-        DB_VER=$(grep spark.databricks.clusterUsageTags.sparkVersion $DB_DEPLOY_CONF | sed -e 's/.*"\(.*\)".*/\1/')
-      # if we did not error out on reads we should have at least four characters "x.y."
-      if (( ${#DB_VER} < 4 )); then
-          echo >&2 "Unable to determine Databricks version, unexpected length of: ${DB_VER}"
-          exit 1
-      fi
-      export PYSP_TEST_spark_databricks_clusterUsageTags_sparkVersion=$DB_VER
-    else
-      cat << EOF
-This node does not define
-- DATABRICKS_RUNTIME_VERSION environment,
-- Files containing version information: $DBR_VERSION, $DB_DEPLOY_CONF
-
-Proceeding assuming a non-Databricks environment.
-EOF
-
-    fi
 
     # Set spark.task.maxFailures for most schedulers.
     #

--- a/jenkins/databricks/common_vars.sh
+++ b/jenkins/databricks/common_vars.sh
@@ -1,6 +1,6 @@
 #!/bin/bash
 #
-# Copyright (c) 2023, NVIDIA CORPORATION. All rights reserved.
+# Copyright (c) 2023-2025, NVIDIA CORPORATION. All rights reserved.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.

--- a/jenkins/databricks/common_vars.sh
+++ b/jenkins/databricks/common_vars.sh
@@ -21,7 +21,52 @@ if [ -n "$EXTRA_ENVS" ]; then
 fi
 
 SPARK_VER=${SPARK_VER:-$(< /databricks/spark/VERSION)}
-export SPARK_SHIM_VER=${SPARK_SHIM_VER:-spark${SPARK_VER//.}db}
+
+
+# Extract Databricks version from deployed configs.
+# spark.databricks.clusterUsageTags.sparkVersion is set automatically on Databricks
+# notebooks but not when running Spark manually.
+#
+# At the OS level the DBR version can be obtailed via
+# 1. DATABRICKS_RUNTIME_VERSION environment set by Databricks, e.g., 11.3
+# 2. File at /databricks/DBR_VERSION created by Databricks, e.g., 11.3
+# 3. The value for Spark conf in file /databricks/common/conf/deploy.conf created by Databricks,
+#    e.g. 11.3.x-gpu-ml-scala2.12
+#
+# For cases 1 and 2 append '.' for version matching in 3XYdb SparkShimServiceProvider
+#
+DBR_VERSION=/databricks/DBR_VERSION
+DB_DEPLOY_CONF=/databricks/common/conf/deploy.conf
+if [[ -n "${DATABRICKS_RUNTIME_VERSION}" ]]; then
+  export PYSP_TEST_spark_databricks_clusterUsageTags_sparkVersion="${DATABRICKS_RUNTIME_VERSION}."
+elif [[ -f $DBR_VERSION || -f $DB_DEPLOY_CONF ]]; then
+  DB_VER="$(< ${DBR_VERSION})." || \
+    DB_VER=$(grep spark.databricks.clusterUsageTags.sparkVersion $DB_DEPLOY_CONF | sed -e 's/.*"\(.*\)".*/\1/')
+  # if we did not error out on reads we should have at least four characters "x.y."
+  if (( ${#DB_VER} < 4 )); then
+      echo >&2 "Unable to determine Databricks version, unexpected length of: ${DB_VER}"
+      exit 1
+  fi
+  export PYSP_TEST_spark_databricks_clusterUsageTags_sparkVersion=$DB_VER
+else
+  cat << EOF
+This node does not define
+- DATABRICKS_RUNTIME_VERSION environment,
+- Files containing version information: $DBR_VERSION, $DB_DEPLOY_CONF
+
+Proceeding assuming a non-Databricks environment.
+EOF
+
+fi
+
+# TODO make this standard going forward
+if [[ "$SPARK_VER" == '3.5.0' ]]; then
+    DB_VER_SUFFIX="${PYSP_TEST_spark_databricks_clusterUsageTags_sparkVersion//./}"
+else
+    DB_VER_SUFFIX=""
+fi
+
+export SPARK_SHIM_VER=${SPARK_SHIM_VER:-"spark${SPARK_VER//.}db${DB_VER_SUFFIX}"}
 
 # Setup SPARK_HOME if need
 if [[ -z "$SPARK_HOME" ]]; then

